### PR TITLE
chore: improved LiveAnnouncer types

### PR DESCRIPTION
**Pull request by anuraghazra**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-aria/live-announcer/src/LiveAnnouncer.tsx [hunk 1]
```diff
@@ -20,17 +20,28 @@ let node = null;
 let clearTimeoutId = null;
 const LIVEREGION_TIMEOUT_DELAY = 1000;
 
+type TAriaLive = 'assertive' | 'off' | 'polite' | undefined;
```
Comment (snowystinger) [inline packages/@react-aria/live-announcer/src/LiveAnnouncer.tsx:23]: we don't use the 'T' naming convention, AriaLive is fine, see https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/interactions/src/useFocusVisible.ts#L21
The aria-live definition in TS is `'aria-live'?: 'off' | 'assertive' | 'polite';` I don't think we need undefined there and we can use the optional when it is truly optional

File: packages/@react-aria/live-announcer/src/LiveAnnouncer.tsx [hunk 2]
```diff
@@ -20,17 +20,28 @@ let node = null;
...
+
+interface IMessageProps {
```
Comment (snowystinger) [inline packages/@react-aria/live-announcer/src/LiveAnnouncer.tsx:25]: we don't use the "I" naming convention
this should be `MessageProps` without the "I" I think

**Review by snowystinger — COMMENTED** [review]
Thanks for taking an interest! This component is in alpha and is subject to change, just a word of warning. It is still good to have types for it though so we can get feedback

File: packages/@react-aria/live-announcer/src/LiveAnnouncer.tsx (hunk 1 above)
Comment (anuraghazra) [reply packages/@react-aria/live-announcer/src/LiveAnnouncer.tsx:23]: Ahh i see okay i'll change these to normal convention. 

🤔 hmm yeah right no need to have undefined in the union. will fix it.

File: packages/@react-aria/live-announcer/src/LiveAnnouncer.tsx (hunk 2 above)
Comment (anuraghazra) [reply packages/@react-aria/live-announcer/src/LiveAnnouncer.tsx:25]: got it!

**Review by dannify — APPROVED** [review]

**Pull request merged**